code/modules/antagonists/traitor/contractor/datums/contractor_hub.dm

```diff
@@ -0,0 +1,184 @@
+			if(difficulty > EXTRACTION_DIFFICULTY_EASY)
+				amount_tc = max(amount_tc, C.reward_tc[difficulty - 1] + (difficulty - 1))
+			C.reward_tc[difficulty] = amount_tc
+			total_earnable_tc[difficulty] += amount_tc
+		// Add to lists
+		contracts += C
+		targets += C.contract.target
+
+	// Fill the gap if a difficulty doesn't meet the TC threshold
+	for(var/difficulty in EXTRACTION_DIFFICULTY_EASY to EXTRACTION_DIFFICULTY_HARD)
+		var/total = total_earnable_tc[difficulty]
+		var/missing = difficulty_tc_thresholds[difficulty] - total
+		if(missing <= 0)
+			continue
+		// Just add the missing TC to a random contract
+		var/datum/syndicate_contract/C = pick(contracts)
+		C.reward_tc[difficulty] += missing
+
+/**
+  * Generates an amount of TC to be used as a contract reward for the given difficulty.
+  *
+  * Arguments:
+  * * total_contracts - The number of contracts being generated.
+  * * difficulty - The difficulty to base the threshold from.
+  */
+/datum/contractor_hub/proc/calculate_tc_reward(total_contracts, difficulty = EXTRACTION_DIFFICULTY_EASY)
+	ASSERT(total_contracts > 0)
+	return CEILING((difficulty_tc_thresholds[difficulty] / total_contracts) * (1 + (rand(-100, 100) / 100) * tc_variation), 1)
+
+/**
+  * Called when a [/datum/syndicate_contract] has been completed.
+  *
+  * Arguments:
+  * * tc - The final amount of TC to award.
+  * * creds - The final amount of credits to award.
+  */
+/datum/contractor_hub/proc/on_completion(tc, creds)
+	completed_contracts++
+	reward_tc_available += tc
+	rep += rep_per_completion
+	owner?.initial_account?.credit(creds, pick(list(
+		"CONGRATULATIONS. You are the 10,000th visitor of SquishySlimes.squish. Please find attached your [creds] credits.",
+		"Congratulations on winning your bet in the latest Clown vs. Mime match! Your account was credited with [creds] credits.",
+		"Deer fund beneficiary, We have please to imform you that overdue fund payments has finally is approved and yuor account credited with [creds] creadits.",
+		"Hey bro. How's it going? You bought me a beer a long time ago and I want to pay you back with [creds] creds. Enjoy!",
+		"Thank you for your initial investment of 500 credits! We have credited your account with [creds] as a token of appreciation.",
+		"Your refund request for 100 Dr. Maxman pills with the reason \"I need way more than 100 pills!\" has been received. We have credited your account with [creds] credits.",
+		"Your refund request for your WetSkrell.nt subscription has been received. We have credited your account with [creds] credits.",
+	)))
+	// Clean up
+	current_contract = null
+
+/**
+  * Gives any unclaimed TC to the given mob.
+  *
+  * Arguments:
+  * * M - The mob to give the TC to.
+  */
+/datum/contractor_hub/proc/claim_tc(mob/living/M)
+	if(reward_tc_available <= 0)
+		return
+
+	// Spawn the crystals
+	var/obj/item/stack/telecrystal/TC = new(get_turf(M), reward_tc_available)
```

Comment:
Any reason you dont just increase the TC in the uplink itself instead of spawning inhand TC?

Reply:
Because that assumes the contractor has the hidden uplink on them, but that may not always be the case. It also makes sharing with other traitors easier that way since there's less steps.